Update the full audio layout on the single result page

obulat opened the pull request:
<!-- prettier-ignore -->
## Fixes
<!-- If PR doesn't fully resolve the issue, replace 'Fixes' below with 'Related to'. -->
<!-- If there is no issue being resolved, please consider opening one before creating this pull request. -->

Related to #2910 by @fcoveram

## Description
<!-- Concisely describe what the pull request does. -->
<!-- Add screenshots, videos, or other media to show the problem and the solution when appropriate. -->
This PR implements the changes to the main audio on the single result from the [updated designs](https://www.figma.com/file/LMGuwErTWIYkOIEZI4t320/Audio-components-improvement?type=design&node-id=1244-54553&mode=design&t=2so3tcWZ9tNVXT8A-4)

## Testing Instructions
<!-- Give steps for the reviewer to verify that this PR fixes the problem; or delete this section entirely. -->


## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. -->

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin
<!-- You must read and understand the following attestation. -->

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

sarayourfriend commented:
> This PR can be approved on its own, but it should be merged together with the other audio layout PRs to prevent the jarring difference between the old and the new layouts.

@obulat would it be possible to put these behind a feature flag, to reduce the deployment complexity or one of these PRs blocking all the others? Only if it wouldn't massively over-complicate things to outweigh avoiding the deployment and PR coordination complexity you've mentioned.

sarayourfriend reviewed (commented):
I like the improvement of this change on the medium width screens, it really is much more sensible. I've looked over the code and it looks good to me, and left a suggestion for how to avoid the PR merging complexity, but will approve once I've had a chance to run the code locally and test it fully.

AetherUnbound reviewed (approved):
This looks much better, and the changes make sense!

sarayourfriend commented:
@obulat any thoughts regarding this? https://github.com/WordPress/openverse/pull/2933#issuecomment-1730444052

I'll re-review and approve as-is if the feature flag doesn't work, but it seems preferable to coordinating various merges. Just let me know either way.

obulat commented on frontend/src/components/VAudioTrack/layouts/VFullLayout.vue:
```diff
@@ -16,12 +16,33 @@
+      :class="[
+        additionalSearchViews
+          ? 'grid grid-cols-1 grid-rows-[auto,auto] p-6 pb-0 lg:mb-6'
+          : 'items-top mt-6 flex flex-row flex-wrap gap-6  px-6 ',
```
```suggestion
          : 'items-top mt-6 flex flex-row flex-wrap gap-6 px-6 ',
```

obulat replied:
```suggestion
          : 'items-top mt-6 flex flex-row flex-wrap gap-6 px-6',
```

obulat commented on frontend/test/playwright/visual-regression/pages/pages-single-result.spec.ts:
```diff
@@ -21,26 +22,53 @@ const cleanRelatedImages = async (page: Page) => {
+      await expectSnapshot(
+        `image-${dir}-from-search-results`,
+        page,
+        {
+          fullPage: true,
+        },
```
```suggestion
        { fullPage: true },
```

fcoveram reviewed (changes requested):
I noticed the following:
* On `xs`, the main CTA is not full width as the mockup. [Screenshot](https://share.cleanshot.com/PHN37cxK).
* On `xs`, the button area doesn't have the right padding bottom value. [Screenshot](https://share.cleanshot.com/vrpttBQs).
* When playing the audio, the audio control button doesn't show the pause icon. [Screenshot](https://share.cleanshot.com/R1nZHvn9).

On a note aside that doesn't block this PR, the full audio component, in both medium and small, [does not have the timer aligned on the corner](https://share.cleanshot.com/ZxLH4wDr) as its other variants do. See this [playing state](https://share.cleanshot.com/wwdsWGXZ) of row large.

Is that possible to change? If not, we can definitely address it in a different ticket.

And finally, audio title has a different text style compared to the mockup, but I'm aware that "Additional search views" project replaces that section. Should this PR address that?

obulat commented:
> I noticed the following:
> 
> * On `xs`, the main CTA is not full width as the mockup. [Screenshot](https://share.cleanshot.com/PHN37cxK).

Updated in [1f6935e](https://github.com/WordPress/openverse/pull/2933/commits/1f6935ef7724500a6a712b19b2c16c9fd2f31348)

> * On `xs`, the button area doesn't have the right padding bottom value. [Screenshot](https://share.cleanshot.com/vrpttBQs).

This was fixed in the previous commits. Maybe you didn't have the updated version of the branch?

> * When playing the audio, the audio control button doesn't show the pause icon. [Screenshot](https://share.cleanshot.com/R1nZHvn9).

This looks like something didn't load correctly. I can see the pause button when I click "Play" locally.

> On a note aside that doesn't block this PR, the full audio component, in both medium and small, [does not have the timer aligned on the corner](https://share.cleanshot.com/ZxLH4wDr) as its other variants do. See this [playing state](https://share.cleanshot.com/wwdsWGXZ) of row large.
> 
> Is that possible to change? If not, we can definitely address it in a different ticket.

Do you mean the time stamp should be higher than it currently is? I don't quite understand what the state should be, and there are no "playing" state mockups (understandably), so I can't see where the timestamp should be, in relation to the yellow background.

> And finally, audio title has a different text style compared to the mockup, but I'm aware that "Additional search views" project replaces that section. Should this PR address that?

I updated the title style, setting it to `heading-6`.

fcoveram reviewed (commented):
I see the previous points correct now.

> Do you mean the time stamp should be higher than it currently is? I don't quite understand what the state should be, and there are no "playing" state mockups (understandably), so I can't see where the timestamp should be, in relation to the yellow background.

I added a mockup in the Figma file. Below is a comparison of the timestamp position with regard to the wave area.

| Current site | Mockup |
|---|---|
| ![Full audio with timestamp area highlighted](https://github.com/WordPress/openverse/assets/895819/6a146d1b-29c5-4979-a865-ffb19d3a31b5) | ![Proposed design of full audio with timestamp area highlighted](https://github.com/WordPress/openverse/assets/895819/a2d6d5cd-e7e0-4eab-9018-8c650cecc5fe) |

The current component has more spacing in the timestamp's top area, whereas the mockup has it aligned in the far top-right corner. This only happens in this component version, the other ones are fine.

obulat commented:
> I added a mockup in the Figma file. Below is a comparison of the timestamp position with regard to the wave area.

Thank you for adding the mockups, I understand now. Can we separate this into a new issue, @fcoveram? It will be difficult to change before updating other Audio track components.

Other than that, I think this PR is ready for re-review.

fcoveram reviewed (approved):
It looks great ✨ 

> Can we separate this into a new issue? It will be difficult to change before updating other Audio track components.

Absolutely 👍
